consensus/safety-rules/src/safety_rules.rs

```diff
@@ -108,7 +140,28 @@ impl<T: Payload> SafetyRules<T> {
             .next_epoch_state()
             .cloned()
             .ok_or(Error::InvalidLedgerInfo)?;
+
+        // By definition, the signer key shall be the same as consensus_key
+        // in case we have an unmatched key, retrieve from the storage the appropriate key
+        match self.verify_key(&epoch_state) {
```

Comment:
probably better to log these where they happen and just have self.verify_key(&epoch_state)?;